modules/local/merge.nf

```diff
@@ -0,0 +1,56 @@
+nextflow.enable.dsl = 2
+
+process MERGE {
+
+    label 'process_low'
+    tag { id }
+
+    container "docker.io/davidfrantz/force:3.7.10"
+
+    input:
+    val (data_type) // defines whether qai or boa is merged
+    tuple val(id), path('input/?/*')
+    path cube
+
+    output:
+    tuple val(id), path("*.tif"), emit: tiles_merged
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    """
+    files=`find -L input/ -type f -printf "%f\\n" | sort | uniq`
+    numberFiles=`echo \$files | wc -w`
+    currentFile=0
+
+    for file in \$files
```

Comment:
I guess same question above from @mashehu about parallelising this with nextflow and not looping this within a module. Or at least if it's lots of very small tasks, you could parallelise with `GNU parallel` or `xargs`